app/src/ui/changes/commit-message.tsx

```diff
@@ -331,15 +336,18 @@ export class CommitMessage extends React.Component<
     )
   }
 
-  private onKeyDown = (event: React.KeyboardEvent<Element>) => {
+  private onKeyDown = (event: React.KeyboardEvent<Element> | KeyboardEvent) => {
     if (event.defaultPrevented) {
       return
     }
 
     const isShortcutKey = __DARWIN__ ? event.metaKey : event.ctrlKey
     if (isShortcutKey && event.key === 'Enter' && this.canCommit()) {
-      this.createCommit()
-      event.preventDefault()
+      const openDialogs = document.getElementsByTagName('dialog')
```

Comment:
We need to check for open foldouts (like the sidebar) here as well. You could check for the presence of an element with id `foldout-container` but I'd rather we proxied `isShowingModal` and `isShowingFoldout` from Repository down to the CommitMessage component and use them instead of looking at the DOM. There's no way for us to know at compile time that this breaks if we inspect the DOM but with a prop we'll know if we attempt to remove it and it won't be tied to the specific markup we choose to use to render dialogs and foldouts.

https://github.com/desktop/desktop/blob/172a9714e17a0365e3e51f3929a7b064d1e9849e/app/src/ui/repository.tsx#L57-L67